### ISSUE TITLE
Reconnect functionality does not appear to work on passworded servers.

### ISSUE THREAD

**Issue by nortonandrews** [Shizmob/pydle#77]
I'm currently connecting a bot to ZNC (which requires a password to be passed to client.connect) and the initial connection works correctly. However, every once in a while this happens:
```
TimeoutError: Ping timeout: no data received from server in 300 seconds.
ERROR:IRCBot:(redacted):Unexpected disconnect. Attempting to reconnect.
WARNING:IRCBot:(redacted):Unknown command: [irc.znc.in] 464 ['(redacted)', 'Password required']
```

The reconnect-fail process repeats itself a few times until finally this exception happens and the bot dies:
```
ERROR:tornado.application:Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x10a8f2840>)
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/tornado/ioloop.py", line 605, in _run_callback
    ret = callback()
  File "/usr/local/lib/python3.6/site-packages/tornado/stack_context.py", line 277, in null_wrapper
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pydle/features/tls.py", line 37, in connect
    return super().connect(hostname, port, tls=tls, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pydle/features/rfc1459/client.py", line 196, in connect
    self._register()
  File "/usr/local/lib/python3.6/site-packages/pydle/features/ircv3/cap.py", line 40, in _register
    super()._register()
  File "/usr/local/lib/python3.6/site-packages/pydle/features/rfc1459/client.py", line 213, in _register
    self.set_nickname(self._attempt_nicknames.pop(0))
IndexError: pop from empty list
```

I haven't looked into it properly but when this function is called during a reconnect, 'password' is always None even if one was initially passed when calling connect() the first time:
https://github.com/Shizmob/pydle/blob/3c1a6d11e949f0f06f4f3e059f859f34e149b4a2/pydle/features/rfc1459/client.py#L188-L196
I'm assuming the password is not being passed along with the rest of the parameters somewhere else in the code.

**Comment by Shizmob** [comment]
Sure sounds like it, this needs a bit of rework. Gonna take a look.

**Comment by theunkn0wn1** [comment]
I see the issue here.

During reconnection, this line [here](https://github.com/Shizmob/pydle/blob/develop/pydle/client.py#L330) calls `connect(reconnect=True)`.

This causes the `password` argument to default to `None` as its not provided.

However, we do have the password stored in `self.password` (which is currently getting overwritten by the functions default argument) so it shouldn't be too hard to implement a check to prevent the overwrite from occurring.

**Comment by theunkn0wn1** [comment]
Issue should be resolved for release v0.9.0
Feel free to re-open if the issue persists
